docs/source/howto/data.rst

```diff
@@ -260,6 +260,359 @@ Finding data
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Suppose you have a database of structures for which you have performed a set of simulations.
+Your goal is to continue the study only for the structures for which the bandgap is higher than ``1.0 eV``
+To achieve that goal you would first need to make a query that finds such structures and puts them into a group.
+Let's go through the whole process step by step.
+
+1. We start with creating the group ``promising_structures`` in the ``bash`` shell:
+
+.. code-block:: console
+
+    $ verdi group create promising_structures
+    Success: Group created with PK = 8 and name 'promising_structures'
+
+2. Then we do a query that finds the structures which have the property ``bandgap`` higher than  ``1.0``.
+
+.. code-block:: python
+
+    qb = QueryBuilder()
+    qb.append(StructureData,  tag='structure', project='*') # Here we are projecting the entire structure object
+    qb.append(CalcJobNode, with_incoming='structure', tag='calculation')
+    qb.append(Dict, with_incoming='calculation', filters={'attributes.bandgap':{'>':1.0}})
+
+3. Finally, we put all the structures into the ``promising_structures`` group.
+
+.. code-block:: python
+
+    group = load_group(label='promising_structures')
+    group.add_nodes(q.all(flat=True))
+
+.. note::
+
+    It is completely fine to add nodes to the group even if they were already there.
+    Hence the steps 2. and 3. can be repeated as is.
+
+To check that the structures were actually added to the group, we can return to the ``bash`` shell and show the group content:
+
+.. code-block:: console
+
+    $ verdi group show promising_structures
+    -----------------  --------------------
+    Group label        promising_structures
+    Group type_string  core
+    Group description  <no description>
+    -----------------  --------------------
+    # Nodes:
+        PK  Type           Created
+    ------  -------------  ----------------
+     37757  StructureData  229D:02h:02m ago
+     25082  StructureData  229D:02h:51m ago
+     41209  StructureData  229D:02h:51m ago
+    119793  StructureData  124D:00h:46m ago
+     36721  StructureData  229D:07h:03m ago
+     12639  StructureData  229D:00h:18m ago
+    129749  StructureData  120D:12h:33m ago
+    105044  StructureData  120D:13h:50m ago
+     55375  StructureData  229D:06h:54m ago
+    103625  StructureData  123D:23h:44m ago
+      1830  StructureData  229D:04h:47m ago
+    113308  StructureData  124D:03h:52m ago
+    101526  StructureData  112D:10h:14m ago
+    100501  StructureData  229D:06h:54m ago
+     63252  StructureData  229D:00h:17m ago
+    112654  StructureData  120D:11h:40m ago
+    115210  StructureData  112D:07h:51m ago
+     70409  StructureData  227D:19h:48m ago
+    103943  StructureData  112D:09h:10m ago
+    104579  StructureData  112D:10h:15m ago
+     73731  StructureData  228D:04h:22m ago
+
+The structures that were added to the group can now be used further.
```

Comment:
I think this is self-evident. So maybe we can provide an example of *how* they could be used further?